stream.go

```diff
@@ -155,6 +160,11 @@ func (cs *clientStream) Trailer() metadata.MD {
 }
 
 func (cs *clientStream) SendMsg(m interface{}) (err error) {
+	cs.mu.Lock()
```

Comment:
This is the thing I always try to avoid -- it adds locking/unlocking overhead even when tracing is disabled. Let's shoot for a way where the fast path is not impacted when tracing is disabled. I am okay to check EnableTracing a couple of times -- it is better than acquiring/releasing locks regardless tracing is enabled or not.